src/ILLink.RoslynAnalyzer/RequiresISymbolExtensions.cs

```diff
@@ -49,18 +49,21 @@ public static bool IsOverrideInRequiresScope (this ISymbol member, string requir
 			return member.IsInRequiresScope (requiresAttribute, false);
 		}
 
-		private static bool IsInRequiresScope (this ISymbol member, string requiresAttribute, bool checkAssociatedSymbol)
+		private static bool IsInRequiresScope (this ISymbol symbol, string requiresAttribute, bool checkAssociatedSymbol)
 		{
-			if (member is ISymbol containingSymbol) {
-				if (containingSymbol.HasAttribute (requiresAttribute)
-					|| (containingSymbol is not ITypeSymbol &&
-						containingSymbol.ContainingType is ITypeSymbol containingType &&
-						containingType.HasAttribute (requiresAttribute))) {
-					return true;
-				}
+			// Requires attribute on a type does not silence warnings that originate
+			// from the type directly. We also only check the containing type for members
+			// below, not of nested types.
+			if (symbol is ITypeSymbol)
+				return false;
+
+			if (symbol.HasAttribute (requiresAttribute)
```

Comment:
I'm thinking a safer thing would be to call "Validate" on the attribute as well - it basically checks that the attribute is in the form we recognize. As a safe guard against potential future changes which could for example add some kind of "feature condition" to the RUC attribute (I know we've discussed this), in which case the attribute is not always "active", so it's safer to NOT recognize it.